### PR TITLE
Implement ShouldBeDecoratedWith() for Type 

### PR DESCRIPTION
**Pull request by hrai**
Resolves and closes https://github.com/shouldly/shouldly/issues/291

### CODE REVIEW

File: src/Shouldly.Tests/TestHelpers/MyDecoratedThing.cs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+namespace Shouldly.Tests.TestHelpers
+{
+    internal class MyDecoratedThing : MyDecoratedBase{ }
```
Comment (sungam3r) [inline src/Shouldly.Tests/TestHelpers/MyDecoratedThing.cs]:
```suggestion
    internal class MyDecoratedThing : MyDecoratedBase { }
```

File: src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            if (!actual.HasAttribute(typeof(T)))
+                throw new ShouldAssertException(new ExpectedShouldlyMessage(typeof(T).GetTypeInfo().Name, () => customMessage).ToString());
+        }
+
+        public static bool HasAttribute(this Type type, Type attributeType)
+        {
+            return type.GetTypeInfo().GetCustomAttributes(attributeType, true).Any();
+        }
+
```
Comment (sungam3r) [inline src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs]:
```suggestion
```

File: src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            ShouldBeDecoratedWith<T>(actual, "");
+        }
+
+        public static void ShouldBeDecoratedWith<T>(this Type actual, string customMessage) where T : class
+        {
+            if (!actual.HasAttribute(typeof(T)))
+                throw new ShouldAssertException(new ExpectedShouldlyMessage(typeof(T).GetTypeInfo().Name, () => customMessage).ToString());
+        }
+
+        public static bool HasAttribute(this Type type, Type attributeType)
```
Comment (sungam3r) [inline src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs]:
```suggestion
        private static bool HasAttribute(this Type type, Type attributeType)
```

File: src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+using System.Linq;
+using System.Reflection;
+
+namespace Shouldly
+{
+    [DebuggerStepThrough]
+    [ShouldlyMethods]
+    public static class ShouldBeDecoratedWithExtensions
+    {
+        public static void ShouldBeDecoratedWith<T>(this Type actual) where T : class
```
Comment (sungam3r) [inline src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs]:
```suggestion
        public static void ShouldBeDecoratedWith<T>(this Type actual) where T : Attribute
```

Same for other methods.

Comment (hrai) [reply src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs]: Updated

**Comment by hrai** [comment]
@sungam3r can you pls see if the comments have been addressed?

**Comment by sungam3r** [comment]
LGTM

**Comment by hrai** [comment]
@sungam3r what's the process for merging PRs. 

Do I need to tag anyone?

**Comment by sungam3r** [comment]
I have no write access to this repo. Ask @JosephWoodward to review/merge PR.

**Comment by hrai** [comment]
@JosephWoodward if you're happy with the changes, can you pls merge it?

**Comment by SimonCropp** [comment]
@JosephWoodward probably easiest for you to merge and resolve the conflicts? or i can resubmit a new PR with the conflicts resolved

**Comment by SimonCropp** [comment]
@hrai thanks

**Pull request merged**